File: Part1_강의자료/Ch08_투포인터/문제별코드/17609_회문/17609.java
import java.util.Arrays;
import java.util.Scanner;

class Main
{
    static boolean isPalindrome(char[] str, int l, int r) {
        while (l <= r) {
            if (str[l] != str[r]) return false;
            l++;
            r--;
        }
        return true;
    }

    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();
        while (T-- > 0) {
            char[] str = sc.next().toCharArray();
            int ans = 0;
            int l = 0, r = str.length - 1;
            while (l <= r) {
                if (str[l] != str[r]) {
                    if (isPalindrome(str, l + 1, r) || isPalindrome(str, l, r - 1)) ans = 1;
                    else ans = 2;
                    break;
                }
                l++;
                r--;
            }
            System.out.println(ans);
        }
    }
}

File: Part1_강의자료/Ch08_투포인터/문제별코드/17609_회문/17609.py
def is_palindrome(s):
    return s == s[::-1]

T = int(input())
for _ in range(T):
    s = input()

    ans = 0
    l, r = 0, len(s) - 1
    while l <= r:
        if s[l] != s[r]:
            if is_palindrome(s[l + 1:r + 1]) or is_palindrome(s[l:r]):
                ans = 1
            else:
                ans = 2
            break
        l += 1
        r -= 1
        
    print(ans)
